test/jdk/java/math/BigDecimal/StringConstructor.java

```diff
@@ -66,7 +66,6 @@ public static void main(String[] args) throws Exception {
         constructWithError("10e"+Integer.MIN_VALUE);
         constructWithError("0.01e"+Integer.MIN_VALUE);
         constructWithError("1e"+((long)Integer.MIN_VALUE-1));
-        constructWithError("1e"+((long)Integer.MAX_VALUE + 1));
 
```

Comment:
Please add some test cases to demonstrate that the round-tripping that previous did not work is functional after the fix.

Reply:
Tests were added in the new commit.